types/time.go

```diff
@@ -653,22 +654,35 @@ func parseDatetime(sc *stmtctx.StatementContext, str string, fsp int, isFloat bo
 				// 20170118.123423 => 2017-01-18 00:00:00
 			} else {
 				// '20170118.123423' => 2017-01-18 12:34:23.234
-				fmt.Sscanf(fracStr, "%2d%2d%2d", &hour, &minute, &second)
+				switch len(fracStr) {
+				case 0:
+				case 1, 2:
+					_, err = fmt.Sscanf(fracStr, "%2d", &hour)
+					// `fmt.Sscanf` parses '1a' into int 1 using format "%d" without raising any error,
```

Comment:
Hi there, how about `fmt.Sscanf(fracStr, "%2d ", &hour)`?

Reply:
``` 

package main

import (
	"fmt"
)

func main() {
	var hour int
	_, err := fmt.Sscanf("1a", "%2d ", &hour)
	fmt.Println(err)
	fmt.Println(hour)
}
```

The output is the following:

```
expected space in input to match format
1
```

Reply:
@zhexuany  interesting....

Reply:
Yeah. The extra space is used to detect the form of `1a`. Without the extra space, It returns 1 and the value of err is nil.